pkg/controllers/vdb/upgrade.go

```diff
@@ -203,9 +204,13 @@ func (i *UpgradeManager) updateImageInStatefulSets(ctx context.Context) (int, er
 func (i *UpgradeManager) updateImageInStatefulSet(ctx context.Context, sts *appsv1.StatefulSet) (bool, error) {
 	stsUpdated := false
 	// Skip the statefulset if it already has the proper image.
-	if sts.Spec.Template.Spec.Containers[names.ServerContainerIndex].Image != i.Vdb.Spec.Image {
+	cnts := sts.Spec.Template.Spec.Containers
+	if !i.Vdb.CheckIfContainersHaveVDBImage(cnts, names.ServerContainerIndex) {
 		i.Log.Info("Updating image in old statefulset", "name", sts.ObjectMeta.Name)
 		sts.Spec.Template.Spec.Containers[names.ServerContainerIndex].Image = i.Vdb.Spec.Image
+		if i.Vdb.IsSideCarDeploymentEnabled() {
+			sts.Spec.Template.Spec.Containers[names.NMAContainerIndex].Image = i.Vdb.Spec.Image
```

Comment:
We update both images together. Is there a case where the server and NMA will use different images? I am trying to understand why we have the `CheckIfContainersHaveVDBImage`. Shouldn't checking the server container be sufficient.

Reply:
Maybe I just overthought. I was thinking about upgrade but even in that case the new container will have the same image. I will make it simpler.